### ISSUE TITLE
OS.get_time_from_unix_time(OS.get_unix_time()) returns wrong month at 0 o clock / sunday

### ISSUE THREAD

**Issue by blurymind** [godotengine/godot#4025]
**Operating system or device:**
Linux, github build from today

**Issue description** (what happened, and what was expected):
it is 0:45 - march  13, sunday, 2016

When I do: print (OS.get_time_from_unix_time(OS.get_unix_time()))
I get: (day:13), (hour:0), (minute:45), (month:2), (second:6), (weekday:0), (year:2016)

When I do: print (OS.get_date())
I get: (day:13), (dst:False), (month:3), (weekday:0), (year:2016)

When I print unix time a few minutes later, I get: 1457830499

Also is weekday 0 sunday? :)

**Comment by Razzlegames** [comment]
Yeah but this seems wrong. Check out this code:

enum Month {
MONTH_JANUARY,
MONTH_FEBRUARY,
MONTH_MARCH,
MONTH_APRIL,
MONTH_MAY,
MONTH_JUNE,
MONTH_JULY,
MONTH_AUGUST,
MONTH_SEPTEMBER,
MONTH_OCTOBER,
MONTH_NOVEMBER,
MONTH_DECEMBER
};

Let me explain the meaning. Enums start at zero implicitly, unless you specify a number for any entry, then it increments by 1 contiguously. Anyhow, this means that MONTH_JANUARY == 0.

This means MONTH_JANUARY != 1 in the code. That's not good IMO.

Maybe I'll fix get_date?  What OS are you using?

**Comment by Razzlegames** [comment]
![screenshot - 03122016 - 06 45 47 pm](https://cloud.githubusercontent.com/assets/8754559/13726399/c24086c4-e882-11e5-80a8-ad8b093fde86.png)

**Comment by Razzlegames** [comment]
Check out this code.  It's totally wrong:

```
var dateDict = OS.get_date()
	if(OS.MONTH_MARCH != dateDict['month']):
		print("month: "+ str(dateDict['month']) + \
		   " but OS.MONTH_MARCH = "+ str(OS.MONTH_MARCH))
```

It means it's essentially not march.  This is broken everywhere actually.

**Comment by Razzlegames** [comment]
I guess the easiest thing is to change the enum to start at 1 and hopefully all the docs get regenerated automatically to reflect that. Also hope there's nowhere else it's inconsistent. So far I see duplicate structures in:

``OS::Month`` ``and OS_::Month``

**Comment by Razzlegames** [comment]
LOL, I don't know what the convention is in Godot, but I am going to go with indexed at 1 since it is in all the get_date implementations and resolve the enums to follow this pattern.  Looks like it was followed from Windows SYSTEMTIME structure: https://msdn.microsoft.com/en-us/library/windows/desktop/ms724950(v=vs.85).aspx

**Comment by Razzlegames** [comment]
All implementations found: 

- OS_Unix::get_date => verified...
- OSNacl::get_date => indirectly verified (see code notes).  This follows unix standard and needed to be +1 for month, to match Julian calendar style 1-12.
- OS_Windows::get_date => Indirectly verified.  SYSTEMTIME already follows 1-12 month format
- OSWinrt::get_date => indirectly verified.  SYSTEMTIME already follows 1-12 month format

**Comment by blurymind** [comment]
@Razzlegames I thanked you on facebook for the help on this a while ago, but now I see that I have overlooked thanking you here personally as well for submitting a patch!

Thank you :+1: 

Do you think we can close this issue now?

**Comment by vnen** [comment]
Yes, this can be closed since it was fixed by #4027.

**Comment by Razzlegames** [comment]
No problem, Todor. I hope I can make some more when I get some free time :)
 Happy Gamedev-ing all
